src/libraries/Native/Unix/System.Security.Cryptography.Native/openssl.c

```diff
@@ -1068,27 +1068,30 @@ int32_t CryptoNative_LookupFriendlyNameByOid(const char* oidValue, const char**
         return -2;
     }
 
+    // First, check if oidValue parses as a dotted decimal OID. If not, we'll
+    // return not-found and let the system cache that.
+    int asnRet = a2d_ASN1_OBJECT(NULL, 0, oidValue, -1);
```

Comment:
The NULL argument tells it to parse it but just throw away the result?

Reply:
Yeah.  It's the good old "tell me how much memory this is going to take" pattern.